webhook/icinga.go

```diff
@@ -137,16 +134,20 @@ func createServiceData(hostname string,
 		// periods are already managed by Prometheus/Alertmanager and relevant
 		// config parameter defaults to 1, but is still tunable for other usecases
 		MaxCheckAttempts: float64(config.MaxCheckAttempts),
+		// Set defaults
+		BridgeUUID: config.UUID,
+		KeepFor:    config.KeepFor,
+		Templates:  config.IcingaConfig.Templates,
 	}
 
 	// Check if this is a heartbeat service. Adjust serviceData
 	// accordingly
 	if heartbeatInterval.Seconds() > 0.0 {
 		l.Infof("Creating alert as heartbeat with check interval %v", heartbeatInterval)
 		// Set dummy text to message annotation on alert
-		serviceData.Vars["dummy_text"] = alert.Annotations["message"]
+		serviceData.DummyText = alert.Annotations["message"]
 		// Set exitStatus for missed heartbeat to Alert's severity
-		serviceData.Vars["dummy_state"] = status
+		serviceData.DummyState = status
```

Comment:
Same here, add these to `serviceVars` instead of having fields on the Service object which only make sense for the `dummy` Icinga2 check.